### ISSUE TITLE
Handle tables and queries API errors on the schema page

### ISSUE THREAD

**Issue by rajatvijay** [centerofci/mathesar#2644]
## Description

The schema page directly relies on two API calls
1. Fetching tables
2. Fetching queries(explorations)

## Current behavior

When either one of the APIs fails the page assumes a success state with no results hence showing the button either create a new table or go to the data explorer to create a new exploration. 
<img width="1510" alt="Screenshot 2023-03-07 at 8 00 55 AM" src="https://user-images.githubusercontent.com/11032856/223310639-b7a3aca9-7bec-41cb-8812-155aa4063c29.png">
 

## Expected behavior
When either one of the APIs fails the page should show an error state with a button to go back to the database page.

**Comment by rajatvijay** [comment]
@kgodey @ghislaineguerin tagging you just in case you want to have a proper designs for this error state.

**Comment by SainiAditya1** [comment]
@kgodey I want to work on this.

**Comment by MANISH-LAB** [comment]
assign me this issue as well, i would love to contribute to it.

**Comment by Anish9901** [comment]
Go ahead @MANISH-LAB.

**Comment by rajatvijay** [comment]
@pavish that was my initial thought too. But then shouldn't these apis have default retry mechanism for let's say 3 times in built?

**Comment by pavish** [comment]
@rajatvijay That's a backend concern (and as far as I know, we don't do anything of that sort) and I don't think we should account for it in the UX. I don't see any specific advantage in having a button move to database page, but I do see an advantage in a retry button, because it lets the user take an action to recover the error state.

**Comment by faraz16iqbal** [comment]
@rajatvijay should be a fairly simple check to check for data from both API calls right? But what should be the desired behavior if the request fails even after retrying? We might also want to rate limit the retry calls to our backend. What do you think?

**Comment by rajatvijay** [comment]
@faraz16iqbal 

> But what should be the desired behavior if the request fails even after retrying?

This is the main reason for having the go-to database page button too. I have already added this in the **Expected** section of the issue description. 


> We might also want to rate limit the retry calls to our backend. What do you think?

I do not think that we should bother ourselves with this since this is similar to the user coming back to this page again and again which ends up making the same no of API calls.

**Comment by monstajoe2002** [comment]
Can I work on this issue?

**Comment by monstajoe2002** [comment]
Do you need a refresh button that makes a GET request?

**Comment by rajatvijay** [comment]
@monstajoe2002 

> Can I be assigned on this issue?

No, but you can start working on this. See https://github.com/centerofci/mathesar/issues/1490#issuecomment-1481147668

> Do you need a refresh button that makes a GET request?

Please see the "Expected behaviour" section of the issue description.

**Comment by monstajoe2002** [comment]
How do I trigger 504 error?

**Comment by Aritra8438** [comment]
> How do I trigger 504 error?

You can send a response with status code 504 from the backend.

**Comment by rajatvijay** [comment]
@Aritra8438 assigning this to you due to in activity from earlier contributor

**Comment by Aritra8438** [comment]
Hi, @rajatvijay, as you have already mentioned, **the error state should have a proper design.**

While submitting the PR, I used a crude template to get things done. After the design, **I will modify the PR accordingly.**

<img width="960" alt="Screenshot 2023-04-20 005210" src="https://user-images.githubusercontent.com/64671908/233178720-0383ba1b-62fa-4f7c-804f-587732817dcc.png">

**Comment by ghislaineguerin** [comment]
@rajatvijay The design is already added in #2829

**Comment by seancolsen** [comment]
Unassigning due to inactivity

**Comment by Tweniee** [comment]
can this be assigned to me?

**Comment by seancolsen** [comment]
@Tweniee You're welcome to work on this and submit a PR, but in general we don't assign tickets to new contributors, as noted in our [Contributor Guide](https://github.com/centerofci/mathesar/blob/develop/CONTRIBUTING.md). Once we've already merged a PR of yours, then we'll be happy to assign tickets to you after that. The reason for this policy is that in the past it has been a maintenance burden for us to manage ticket assignments while having an overwhelming volume of people fail to submit their first PR when initially expressing intent to contribute.

**Comment by nikhilhenry** [comment]
Hi, I have created PR #3323 to address this issue. 

Could this issue please be assigned to me?

Thanks :)

**Comment by pavish** [comment]
@nikhilhenry As noted in our [Contributor Guide](https://github.com/mathesar-foundation/mathesar/blob/develop/CONTRIBUTING.md) and in the previous comment by Sean, we assign new issues to contributors after one of their PRs are merged.

Thanks for the PR #3323. After we've reviewed and merged it, if you wish to work on other open issues, we'd happily assign them to you.
